src/UploadedFile.php

```diff
@@ -102,9 +103,13 @@ public function __construct(
         if ($errorStatus === UPLOAD_ERR_OK) {
             if (is_string($streamOrFile)) {
                 $this->file = $streamOrFile;
+
+                parent::__construct($this->file);
             }
             if (is_resource($streamOrFile)) {
                 $this->stream = new Stream($streamOrFile);
+
+                parent::__construct($this->stream->getMetaData('uri'));
```

Comment:
Does `SplFileInfo` works with stream paths? PHP docs say the argument for `SplFileInfo::__construct()` should be a file path while the value of `$this->stream->getMetaData('uri')` could be a stream path `'php://memory'`.

As commented by @michalbundyra this patch should have more tests.

Reply:
As suspected things go kaboom when trying to use some of the `SplFileInfo` methods if it's created with a stream path.

https://3v4l.org/klK0X

So I am not sure if this is a very good change.

Reply:
@ADmad
Thanks for your comment! 👍 

We have another discussion in the related issue report: #377

Reply:
@ADmad nowo run getMTime() on a file that doesn't exist... SplFileInfo is doing exactly what it is supposed to do here.  This is not going "kaboom" -- this is how it works.  It's not intended to provide information about files that it can't get information for -- which includes any non existent file.  Your test seems to be based on a false assumption about SplFileInfo that it requires an actually existing file... it doesn't.  SplFileInfo is designed to grok information about files, which means the path you pass it MAY not exist, and it's up to a developer to use it properly.

There may, of course be edge cases that we'd want to handle differently.  But the example provided is just SplFileInfo working as it does.

Reply:
> SplFileInfo is designed to grok information about files

Exactly, "about files" not streams.

Reply:
@mattsah —

Having the class extend `SplFileInfo` implies that the item returned has a relationship with the filesystem, and can be operated as if it were part of the filesystem. It also implies that users should be able to safely call its various methods.

Considering the major use case is to use a temporary stream (Diactoros creates temporary streams by default), we absolutely need stream support to work without errors. Since it wasn't, we had to revert.

Reply:
@weierophinney 

Is there a single case where SplFileInfo acts differently on a stream than it does on a non-existent file?  It seems quite strange to me that Spl**File**Info suggests a relationship with the filesystem while Uploaded**File** does not.  A stream has the same relationship to the filesystem as a non-existent file.  Again, the implication here seems to be that SplFile**Info** is not about grokking information.

Additionally, if one is truly concerned with supporting streams more fully, it would be perfectly fine, for example, to overload getMTime() later and call parent if its a file or examine stream meta data if it's a stream.

PHP uses stream URLs in places of "files" in all sorts of cases... the implication you suggest doesn't seem well supported.

Regarding major use cases and defaults... given that this did not previously extend it, I don't see any case where anyone would be relying on SplFileInfo methods... stream support would absolutely continue working as it has historically and **could be improved upon** to offer even more info via overloads as mentioned above in the future.

This is non-breaking, working as expected, and providing a useful feature.  It may not provide all the features you want out of the gate, i.e. functional mtime on stream, but that'd make for a nice 2.2.1 -- awell.

Very disappointed here.